### DISCUSSION TITLE
Roadmap for UI development

### DISCUSSION THREAD

**Discussion by mzhubail** [firefly-iii/firefly-iii#9145] [General]
From the project's [website](https://docs.firefly-iii.org/references/firefly-iii/api/), I saw that there is a desire for using the API exclusively for the official UI.

> The API is slated to become the only way to talk to Firefly III: even the UI would use the API exclusively. Alas, this is a work in progress.

I'm interested in contributing to this, as well as other additions.

From what I can see the UI is written in twig, blade, and vue.  (A) Is there's a specific reason for this? and (B) Is there an interest to partially change to a unified way of making the UI (e.g. use blade exclusively)?

In addition to that, there are some minor UI enhancements in mind: like making the sidebar scroll independently of the content of the page.

I just wanted to see what the project's author has in mind and what of the stuff I have in mind are relevant.

**Comment by JC5** [comment]
The current layout is written in Twig, Vue and plain Javascript. The new one will be written in Blade and AlpineJS. So yes, this will be unified, but won't be done in one go.

As improvements go, the current (`v1`) layout is pretty much frozen, and I try to spend all my time on the new layout. What did you have in mind?

**Reply by tasnim0tantawi** [reply]
Can we help in migrating the UI and adding design enhancements? What we have in mind right now is: 
1. Enhancing the cards design. 
![image](https://github.com/user-attachments/assets/7d30b212-1e3e-4af9-91cf-ea2e4a8ed457)

2. Making the + button more visible.
![image](https://github.com/user-attachments/assets/f2b0a367-a506-4797-9a5d-2d9b63507bef)

3. Fix scrolling menu in mobile view.
4. Arrow directions. 
![image](https://github.com/user-attachments/assets/8ea023e2-03d0-4a34-a33f-3ea08026a91f)
5. Adding front end validation (before sending the request)/

**Reply by tasnim0tantawi** [reply]
@JC5 profile page and adding more languages as well.

**Reply by JC5** [reply]
Hey,

Feel free to submit a PR for 1, 2, 3 and 4, I'm curious to see what you have in mind. Front-end validation (5) isn't going to be added, too much work for too little gain. 

What do you have in mind for the profile page and "adding more languages"?

**Reply by mzhubail** [reply]
@JC5 If I may ask about the new layout, is it "new" as in new user interface, or "new" because current user interface will be migrated to blade and AlpineJS?

I'm still trying to wrap my head around the project structure.  Is `resources/assets/v1` the old frontend, and `resources/assets/v2` the new one with blade and alpine only?

**Reply by tasnim0tantawi** [reply]
We would like to add Arabic locale. For profile, we would like to create a nice card layout for it.

**Reply by JC5** [reply]
Yes, v1 is the old one, v2 is the new one. You can try for yourself: `FIREFLY_III_LAYOUT=v1` or `FIREFLY_III_LAYOUT=v2`.

You can skip the card layout, it's not necessary for the old layout.

As for the locales, they are already available. If you want to do translations, head over to crowdin: https://crowdin.com/project/firefly-iii/
